### PR TITLE
Reinstate Test-Connection tests

### LINKED ISSUE #13309: The 'TraceRoute' test take 44 minutes to fail in macOS CI
The test `TraceRoute works` in the context `TraceRoute` in the file `Test-Connection.Tests.ps1` is causing timeout in [macOS CI builds](https://dev.azure.com/powershell/PowerShell/_build/results?buildId=58605&view=logs&jobId=dd0d77e4-0911-5114-ec89-d447d00ba5e6&j=dd0d77e4-0911-5114-ec89-d447d00ba5e6&t=b551df83-c97f-5a24-b379-7e608cc7f18b).

image

We will make the test pending for further investigation.

### PR DESCRIPTION
**Pull request by rjmholt**
# PR Summary

Fixes https://github.com/PowerShell/PowerShell/issues/13309.

Reverts some of the changes in https://github.com/PowerShell/PowerShell/pull/12943.

The `Test-Connection` tests first failed in CI due to a dependency on CI system DNS resolution. We tried to address that in #12943 but recently hit a failure where the new form of the tests using the gateway address also failed consistently across builds.

Given that the DNS issue was resolved, this changes the tests that have been failing back to using the hostname.

There's also a test that didn't work with DNS, was working with the gateway address, but is not working now. That test is now marked as pending.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by rjmholt** [comment]
@PaulHigin I'm seeing a remoting test failure here in the elevated Windows tests I think:

```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Management.Automation.Remoting.PrioritySendDataCollection.ReadOrRegisterCallback(OnDataAvailableCallback callback, DataPriorityType& priorityType) in D:\a\1\s\src\System.Management.Automation\engine\remoting\fanin\PriorityCollection.cs:line 240
   at System.Management.Automation.Remoting.Client.WSManClientCommandTransportManager.SendOneItem() in D:\a\1\s\src\System.Management.Automation\engine\remoting\fanin\WSManTransportManager.cs:line 3975
   at System.Management.Automation.Remoting.Client.WSManClientCommandTransportManager.OnRemoteCmdSendCompleted(IntPtr operationContext, Int32 flags, IntPtr error, IntPtr shellOperationHandle, IntPtr commandOperationHandle, IntPtr operationHandle, IntPtr data) in D:\a\1\s\src\System.Management.Automation\engine\remoting\fanin\WSManTransportManager.cs:line 3681
An error has occurred that was not properly handled. Additional information is shown below. The PowerShell process will exit.
Execution of {& $powershell $PSFlags -c $command} by build.psm1: line 1332 failed with exit code -1073741819
At D:\a\1\s\build.psm1:2192 char:17
+                 throw $errorMessage
+                 ~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (Execution of {& $po…it code -1073741819:String) [], RuntimeException
+ FullyQualifiedErrorId : Execution of {& $powershell $PSFlags -c $command} by build.psm1: line 1332 failed with exit code -1073741819
##[error]PowerShell exited with code '1'.
```

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 1]
```diff
@@ -44,6 +44,12 @@ Describe "Test-Connection" -tags "CI" {
         $targetAddress = "127.0.0.1"
         $targetAddressIPv6 = "::1"
         $UnreachableAddress = "10.11.12.13"
+        # this resolves to an actual IP rather than 127.0.0.1
```
Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:47]: I don't understand why this commented out script is here.  Is this something we want to revert to, or avoid?

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 2]
```diff
@@ -105,7 +111,7 @@ Describe "Test-Connection" -tags "CI" {
             $error[0].Exception.InnerException.ErrorCode | Should -Be $code
         }
 
-        It "Force IPv4 with implicit PingOptions" -Pending {
+        It "Force IPv4 with implicit PingOptions" {
```
Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:114]: Please add comment as to why it is marked 'pending'.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 (hunk 2 above)
Comment (vexx32) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:114]: The pending was removed from this test?

**Review by vexx32 — APPROVED** [review]
LGTM apart from one of Paul's comments, we should either remove or comment as to why the extraneous code is left in. 🙂

**Comment by rjmholt** [comment]
The macOS CI is still failing for the minute, so this doesn't seem to solve our problems with these tests. I'll need to find more time this week to work on this

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**